File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.cpp
class Solution {
public:
    vector<string> buildArray(vector<int>& target, int n) {
        stack<string> st;
        for(int i = 1; i <= n ; i++)
        {
            if(count(begin(target),end(target), i) > 0) st.push("Push");
            else if(target[target.size() -1] < i) break;
            else
            {
                st.push("Push");
                st.push("Pop");
            }
            
        }
        vector<string> v;
        while(st.size())
        {
            v.push_back(st.top());
            st.pop();
        }
        reverse(begin(v),end(v));
        return v;
    }
};

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, t: List[int], n: int) -> List[str]:
        st = []
        for i in range(1,n+1):
            if t.count(i) > 0:
                st.append("Push")
            elif i > t[-1]:
                break
            else:
                st.append("Push")
                st.append("Pop")
        return st